test/e2e/karmadactl_test.go

```diff
@@ -599,22 +602,22 @@ var _ = framework.SerialDescribe("Karmadactl cordon/uncordon testing", ginkgo.La
 			}
 			err := karmadactl.RunCordonOrUncordon(karmadactl.DesiredUnCordon, karmadaConfig, opts)
 			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-			ginkgo.By(fmt.Sprintf("cluster %s taint will be removed", clusterName), func() {
+
+			ginkgo.By(fmt.Sprintf("cluster %s taint(unschedulable:NoSchedule) will be removed", clusterName), func() {
 				clusterObj := &clusterv1alpha1.Cluster{}
 				err := controlPlaneClient.Get(context.TODO(), client.ObjectKey{Name: clusterName}, clusterObj)
 				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-				gomega.Expect(clusterObj.Spec.Taints).Should(gomega.BeEmpty())
-			})
 
-			framework.CreatePropagationPolicy(karmadaClient, policy)
-			framework.CreateDeployment(kubeClient, deployment)
-			ginkgo.DeferCleanup(func() {
-				framework.RemoveDeployment(kubeClient, deployment.Namespace, deployment.Name)
-				framework.RemovePropagationPolicy(karmadaClient, policy.Namespace, policy.Name)
-			})
-			ginkgo.By(fmt.Sprintf("deployment will schedule to cluster %s", clusterName), func() {
-				targetClusters := framework.ExtractTargetClustersFrom(controlPlaneClient, deployment)
-				gomega.Expect(targetClusters).Should(gomega.ContainElement(clusterName))
+				taints := clusterObj.Spec.Taints
+				var unschedulableTaint *corev1.Taint
+				for index := range taints {
+					if taints[index].Key == clusterv1alpha1.TaintClusterUnscheduler &&
+						taints[index].Effect == corev1.TaintEffectNoSchedule {
+						unschedulableTaint = &taints[index]
+						break
+					}
+				}
```

Comment:
I guess we can use [TaintExists](https://github.com/karmada-io/karmada/blob/ac2e9053561647fee7c854b2600f0c543c999c37/pkg/util/helper/taint.go#L16) here, for code readability.